src/tenable/TenableClient.ts

```
@@ -287,7 +287,7 @@ export default class TenableClient {
           return retryDelay;
         },
         handleError: (err, context) => {
-          if (err.statusCode !== 429 && err.statusCode !== 500) {
+          if (![429, 500, 504].includes(err.statusCode)) {
             context.abort();
```

(Placement of this comment is unrelated to the comment)Do we maybe not want to throw an error if we hit max retries on some of this stuff? Perhaps just skip that record and move on or skip the step and move on?

---

I depends...

* If the request is for a list of things, or a page of the list of things, then an incomplete set would lead to flapping data in the graph, which could produce alerts for the J1 account if they're monitoring the presence/absence of any data. We can avoid deletes by failing the step with an error. We have talked about the need for steps to have the ability to communicate partial datasets without raising an error. Until we can do that, for intermittent errors, it is best to throw the error.
* If the request is for an individual item (more details about an item in a list), the item will be incomplete, but it should be okay to send up the incomplete data because only when a property is set to `null` does it delete the property from the entity we already have in the graph. In this case, we should avoid throwing the error.